### ISSUE TITLE
esc doesn't run normally sometimes after using this cool configuration

### ISSUE THREAD

**Issue by zhangjun-xyz** [GoldenChaos/GoldenChaos-BTT#13]
It's awesome and really improve my working efficiency on touch bar. But what annoys me sometimes is that esc button on the touch bar doesn't react. As I am a vim user, I really need this button. Did I do something wrong?

**Comment by dducro** [comment]
Not sure what's wrong with your esc button..

However, I did the following to enable a physical esc button.
I personally don't use the § button below esc.
So I mapped § to esc in BTT.
As well ass option+command+§ to option+command+esc to open the force quit menu.

Hope it helps!

**Comment by zhangjun-xyz** [comment]
Sure that's really an option. 
I also have another problem. I think it's really cool but the problem is I cannot use many app's touch bar functionalities. How to solve this problem? Thanks a lot.

**Comment by dducro** [comment]
I'm not a dev on GoldenChaos-BTT, so I'm not sure on how to solve your issues.
However It would help devs on diagnosing the issue with more details, e.g. screenshots, logs, steps on how to reproduce your issue and what you expected to happen.

**Comment by zhangjun-xyz** [comment]
@dducro Thank you so much for your quick replies. 
To reproduce the problem is so easy, because once you installed this cool configuration you cannot use system's touch bar functionalities anymore.

**Comment by dducro** [comment]
@zhangjun-xyz Haha, true!
I personally never use the native touch-bar.

**Comment by dducro** [comment]
You can pause BTT entirely for specific apps.
Then you can use the native touch-bar, for those apps and use GoldenChaos-BTT for the rest.
(for example, if you need the function keys in one app, I use fn to toggle the function keys)

**Comment by zhangjun-xyz** [comment]
That's really a good idea. I ignored this configuration. Cool. And I think another option is to copy some configurations from GoldenChaos-BTT and make a new one. I don't wanna it too heavy.

**Comment by moblabdev** [comment]
I'm running into the same problem as @zhangjun-xyz too. When using GoldenChaos-BTT, my physical escape button no longer works (particularly painful for vim). I've also mapped caps lock to escape via the Modifier Keys in the keyboard settings and that does not work either.

**Comment by GoldenChaos** [comment]
Apologies for taking so long to respond - is the issue that GC is messing with the hardware esc key on the new Macs? If so, I’m very interested in more info about this since I don’t own one of the new MacBooks. GC doesn’t change anything about how esc keys work, but if there is an underlying BTT issue here it’ll be good to get it fixed!

**Comment by dducro** [comment]
Hey, back on this old issue.
The esc key just stopped working for me when changing the "Use Normal Escape Key" open in GC-BTT Settings.
After this, all esc keys (also from an external keyboard) stopped responding when I have toggle this setting on or off, the esc key stays visible in the touchbar.
(before I only had the fullscreen button visible)
@GoldenChaos Could you revisit this issue? I'll provide you as much debugging info as needed.

**Comment by KyleU** [comment]
Yep, when I moved to a new MacBook with a hardware escape key, installing GoldenChaos prevents the escape key from working. I've tried reinstalling BTT and GC.

Interestingly, the on-screen keyboard's "esc" doesn't work either when GC is running

**Comment by ckjeter** [comment]
issue still remain even with physical esc on external keyboard. I need to disable GC every time I use VIM.

**Comment by SaneBow** [comment]
Same issue with external keyboard. Need to press and hold ESC to make it work sometimes. Then I need to restart BTT to make it resume to normal (where tap ESC works).
